TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyDecodedHttpRequestHandler.java

```diff
@@ -28,59 +27,79 @@ public class NettyDecodedHttpRequestHandler extends ChannelInboundHandlerAdapter
-                    pipelineOrchestrator.addBaselineHandlers(pipeline);
-                    ctx.fireChannelRead(httpJsonMessage);
-                    pipelineOrchestrator.removeThisAndPreviousHandlers(pipeline, this);
-                    return;
-                } else {
-                    pipelineOrchestrator.addContentRepackingHandlers(pipeline);
-                }
+                handlePayloadNeutralTransformation(ctx, request, httpJsonMessage, pipelineOrchestrator);
             } catch (PayloadNotLoadedException pnle) {
+                log.debug("The transforms for this message require payload manipulation, " +
+                        "all content handlers are being loaded.");
                 // make a fresh message and its headers
                 httpJsonMessage = parseHeadersIntoMessage(request);
                 pipelineOrchestrator.addJsonParsingHandlers(pipeline, transformer);
+                ctx.fireChannelRead(httpJsonMessage);
             }
-            // send both!  We'll allow some built-in netty http handlers to do their thing & then
-            // reunify any additions with our headers model before serializing
-            ctx.fireChannelRead(request);
-            ctx.fireChannelRead(httpJsonMessage);
         } else if (msg instanceof HttpContent) {
-            if (msg instanceof LastHttpContent) {
-                statusWatcher.accept(HTTP_CONSUMPTION_STATUS.DONE);
-            }
             ctx.fireChannelRead(msg);
         }
     }
 
+    private void handlePayloadNeutralTransformation(ChannelHandlerContext ctx,
+                                                    HttpRequest request,
+                                                    HttpJsonMessageWithFaultablePayload httpJsonMessage,
+                                                    RequestPipelineOrchestrator pipelineOrchestrator)
+            throws PayloadNotLoadedException
+    {
+        var pipeline = ctx.pipeline();
+        transformer.transformJson(httpJsonMessage);
+        if (headerFieldsAreIdentical(request, httpJsonMessage)) {
+            log.info("Transformation isn't necessary.  " +
+                    "Clearing pipeline to let the parent context redrive directly.");
+            while (pipeline.first() != null) {
+                pipeline.removeFirst();
+            }
+        } else if (headerFieldIsIdentical("content-encoding", request, httpJsonMessage) &&
+                headerFieldIsIdentical("transfer-encoding", request, httpJsonMessage)) {
+            log.info("There were changes to the headers that will be reformatted through this pipeline, " +
+                    "but the content (payload) doesn't need to be transformed.  Content Handlers are not " +
+                    "being added to the pipeline");
+            pipelineOrchestrator.addBaselineHandlers(pipeline);
+            ctx.fireChannelRead(httpJsonMessage);
+            pipelineOrchestrator.removeThisAndPreviousHandlers(pipeline, this);
```

Comment:
Not relevant for right now, but would like to discuss at some point what the solution looks like if we avoid removing handlers and have some sort of skipping mechanism for handlers that we don't need or if there is any way to detect the needed handlers sooner. I think this story is a little easier to tell if the pieces aren't dynamically changing

Reply:
This part of the story is easier to tell - and even writing the logic to just pass stuff through isn't terrible.  Where we'll suffer most might be in logging.  You'd see way more flowing through the handlers with logging turned up, even though the data was static.